src/lineSegment.js

```diff
@@ -43,10 +43,95 @@ var cornerstoneMath = (function (cornerstoneMath) {
         return Math.sqrt(distanceToPointSquared(lineSegment, point));
     }
 
+    // Returns intersection points of two lines
+    function intersectLine(lineSegment1, lineSegment2) {
+        var intersectionPoint = {};
+
+        var x1 = lineSegment1.start.x, y1 = lineSegment1.start.y, x2 = lineSegment1.end.x, y2 = lineSegment1.end.y,
+            x3 = lineSegment2.start.x, y3 = lineSegment2.start.y, x4 = lineSegment2.end.x, y4 = lineSegment2.end.y;
+
+        var a1, a2, b1, b2, c1, c2; // Coefficients of line equations
+        var r1, r2, r3, r4; // Sign values
+
+        var denom, offset, num; //Intermediate values
+
+        // Compute a1, b1, c1, where line joining points 1 and 2 is "a1 x  +  b1 y  +  c1  =  0"
+        a1 = y2 - y1;
+        b1 = x1 - x2;
+        c1 = x2 * y1 - x1 * y2;
+
+        // Compute r3 and r4
+        r3 = a1 * x3 + b1 * y3 + c1;
+        r4 = a1 * x4 + b1 * y4 + c1;
+
+        /* Check signs of r3 and r4.  If both point 3 and point 4 lie on
+         * same side of line 1, the line segments do not intersect.
+         */
+
+        if (r3 !== 0 &&
+            r4 !== 0 &&
+            cornerstoneMath.sign(r3) === cornerstoneMath.sign(r4)) {
+            intersectionPoint.x = 0;
+            intersectionPoint.y = 0;
+            intersectionPoint.intersected = false;
+            return intersectionPoint;
+        }
+
+        /* Compute a2, b2, c2 */
+
+        a2 = y4 - y3;
+        b2 = x3 - x4;
+        c2 = x4 * y3 - x3 * y4;
+
+        /* Compute r1 and r2 */
+
+        r1 = a2 * x1 + b2 * y1 + c2;
+        r2 = a2 * x2 + b2 * y2 + c2;
+
+        /* Check signs of r1 and r2.  If both point 1 and point 2 lie
+         * on same side of second line segment, the line segments do
+         * not intersect.
+         */
+
+        if (r1 !== 0 &&
+            r2 !== 0 &&
+            cornerstoneMath.sign(r1) === cornerstoneMath.sign(r2)) {
+            intersectionPoint.x = 0;
+            intersectionPoint.y = 0;
+            intersectionPoint.intersected = false;
```

Comment:
Can you remove the 'intersected' attribute and just return undefined if there is no intersection?

This function should just return a point object with x and y coordinates, or undefined if no intersection exists.